DataAccess/BookDAO.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Group2_BookStore.DB;
+using Group2_BookStore.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace DataAccess
+{
+    public class BookDAO
+    {
+       private readonly BOOKSTOREContext context; 
+        public BookDAO(BOOKSTOREContext _context) {
+            this.context = _context;
+        }
+        public IEnumerable<Book> GetBookList()
+        {
+            var Books = new List<Book>();
+            try
+            {
+                Books = context.Books.ToList();                
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(ex.Message);
+            }
+            return Books;
+        }
+
+        public IEnumerable<Book> GetBookListByCate(String category)
+        {
+            var Books = new List<Book>();
+            try
+            {
+                Books = context.Books.ToList();                
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(ex.Message);
+            }
+            return Books;
+        }
+        public Book GetBookById(int BookId)
+        {
+            Book Book = null;
+            try
+            {
+                Book = context.Books.SingleOrDefault(c => c.BookId == BookId);
```

Comment:
Chỗ này ông khai báo thẳng luôn, khỏi khai báo ở trên
```suggestion
                var Book = context.Books.SingleOrDefault(c => c.BookId == BookId);
```